### ISSUE TITLE
Firefox crashes when opening file picker on first launch

### ISSUE THREAD

**Issue by Soundtoxin** [NixOS/nixpkgs#87787]
**Describe the bug**
In certain cases, opening the file picker (whether to save an image or upload an image to a file host) will freeze the browser for a few seconds until it crashes.

**To Reproduce**
1. Launch Firefox
2. Browse to a file host such as https://catbox.moe/
3. Attempt to upload a file.

**Expected behavior**
Needless to say, I expect basic functionality to not freeze and crash the whole browser.

**Additional context**
I'm running Sway on Wayland. Firefox version 75.
This seems to happen pretty consistently after a fresh boot + browser launch. If I just rebooted, launched Firefox, and then try to use the file picker, I get a crash. After re-launching the browser, it tends to work normally afterward. I had a day recently where crashing was constant, but I can't re-create this. I think it was something like Sway had crashed and then I re-launched it from the same TTY instead of a new one, and in that session Firefox was buggier than usual, even having crashes when pasting things.

**Notify maintainers**

@edolstra 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.118, NixOS, 20.09pre225264.683c68232e9 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixos-20.09pre225264.683c68232e9"`
 - channels(brad): `"home-manager, nixos-20.09pre225264.683c68232e9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: firefox
# a list of nixos modules affected by the problem
module:
```

**Comment by evils** [comment]
can you try this with firefox-wayland-76?

**Comment by Soundtoxin** [comment]
Yes, I'll do some testing when it's next convenient for me to reboot. I'm now on Firefox 76 instead of 75, so I'll see if this crashes on first launch after a reboot, and then next install firefox-wayland to test.
Do I have to remove firefox to install firefox-wayland, or can they co-exist? I am also using the `export MOZ_ENABLE_WAYLAND=1` environment variable via my `.zprofile` already, by the way.

**Comment by evils** [comment]
i'm given to understand setting that env var should be equivalent to using firefox-wayland  
so while i can't offer a solution, trying it with the latest package is always a good idea (we're at 76.0.1 by now)

**Comment by Soundtoxin** [comment]
Okay, if it's really the same as that env var, I probably won't bother installing firefox-wayland.
>trying it with the latest package is always a good idea (we're at 76.0.1 by now)


I run NixOS Unstable and run updates every day, sometimes multiple times. I am getting Firefox via home-manager, though, so I'm not sure if that somehow puts me slightly behind.
It does appear I'm on 76.0.1 now, so no worries there.
I'll try to test stuff now this weekend. I've got some unfinished stuff open so I just want to do some other things before I next reboot, and that's still the only way I know to easily trigger the problem from before.

**Comment by Soundtoxin** [comment]
Okay, I rebooted and did everything as I usually do, including launching Firefox from rofi, even though I was tempted to launch from a terminal in case it'd have helpful errors there. No crashes! I tried both uploading a file and saving a file, as both had caused a crash before. I felt like I noticed a bit of delay the first time the file picker spawned, but no crash. I guess for now I'm gonna say it was probably fixed in one of the last couple updates.
Closing issue.
